### ISSUE TITLE
Possible bug regarding AI

### ISSUE THREAD

**Issue by assaultboy** [usecforce/USEC_IssuesWiki#47]
If you spawn AI via scripts local to the server they are not able to damage players. However if spawned local to a client they function as intended.

To replicate this enter zeus on the server and execute this on the server:

```
private ["_grp"];

_grp = createGroup EAST;

"O_Soldier_TL_F" createUnit [[9561,9881,0],_grp];
```

It will spawn a CSAT unit in the spawn area, he will be unable to damage any players.

However if you execute the script on the local machine the CSAT unit will be able to damage players.

**Comment by jonpas** [comment]
Can you test using Vanilla and if that works try to find out which mod breaks it?

It's definitely a huge problem we should look into now before running into issues in the future.

**Comment by assaultboy** [comment]
I'm not sure how to go about testing this in vanilla as I don't have a server.

**Comment by jonpas** [comment]
How exactly did you find out about the error then?

**Comment by assaultboy** [comment]
Making my automated bonyo on the paris training server. I had a function to spawn in the enemies for the waves but I noticed they couldn't hurt me. I got around it by spawning the AI on the client that triggers the wave start.

**Comment by jonpas** [comment]
Could be some BI bug, someone needs to test at some point.

How exactly did you run it on Training (Paris) btw?

**Comment by assaultboy** [comment]
Ares has an execute code module. It lets you run on all machines/Local/Server

**Comment by jonpas** [comment]
Aha, that could very well be the cause of it all. I don't like that module at all.

**Comment by assaultboy** [comment]
Shhh. It's the best module.

**Comment by jonpas** [comment]
No it's not, it exposes huge vulnerabilities and apparently doesn't work as it should. :grin:

**Comment by assaultboy** [comment]
Yeah, but it adds a massive amount of capability that I think outweighs the vulnerabilities (zBonyo for example). And as far as I'm aware it works fine, I've never had a problem with it.

**Comment by jonpas** [comment]
Yeah sure, I don't oppose that (except that zBonyo probably doesn't fit Training server, so please make sure it never intrudes training, you can run it on Fun server when we get that afterwards as you like).

**Comment by assaultboy** [comment]
Yeah, I'm waiting for the fun server :P. But until then training's all we got. Plus the Aussies just got there own server up and running so we have that now aswell.

**Comment by jonpas** [comment]
> Yeah, I'm waiting for the fun server :P. But until then training's all we got.

Yes, but that doesn't mean anyone can do whatever they want on that server, if Training is going on that has 100% priority, no questions asked.

> Plus the Aussies just got there own server up and running so we have that now aswell.

I was not aware of that and it is not an official server so I can't speak for it.

**Comment by assaultboy** [comment]
> Yes, but that doesn't mean anyone can do whatever they want on that server, if Training is going on that has 100% priority, no questions asked.

Agreed.

As for the Aussie server, I guess talk to Stark and Aberhart. It's a really nice server.
`221.121.157.135:2302`

**Comment by jonpas** [comment]
I am for sure not going to contact them, I think I do way too much for USEC members to at least justify them contacting me if they want something (eg. make it official or add them to the list of "server maintainers", because if someone reports a bug from that server I will go yell at them :grin: ).

It's great people stepped up and are actually providing a server though, but as said I can't care about it unless they want me to for a good reason.

**Comment by SyMP2005** [comment]
On the regard of no damage, Zeus was unable to damage friendlies with simulated mortar fire (for training purpose). Perhaps there is a larger issue here.

**Comment by jonpas** [comment]
> Zeus was unable to damage friendlies with simulated mortar fire (for training purpose).

Isn't "simulated" meant to mean it doesn't actually damage?

And yes, probably a large issue, but I suggest waiting until next modpack update and checking again.

**Comment by SyMP2005** [comment]
No by "simulated" I meant that Zeus created mortar fire them instead of an actual mortar somewhere.

**Comment by jonpas** [comment]
Ok, thanks for the info (and I have no idea what could cause it).
